tools/mac_setup.sh

```diff
@@ -81,6 +81,9 @@ export CPPFLAGS="$CPPFLAGS -I${BREW_PREFIX}/opt/openssl@3/include"
 export PYCURL_CURL_CONFIG=/usr/bin/curl-config
 export PYCURL_SSL_LIBRARY=openssl
 
+# generate macos requirements.txt
+$ROOT/scripts/uvlock.sh --no-update
```

Comment:
this will generate a diff on those machines and defeats the purpose of the lock. on unsupported platforms, we can just install without using the lock